cmd/trt/main.go

```diff
@@ -0,0 +1,283 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"sort"
+	"strings"
+	"sync"
+
+	"github.com/hashicorp/go-version"
+	"github.com/openshift/ci-tools/pkg/api"
+	"github.com/openshift/ci-tools/pkg/config"
+	"github.com/openshift/ci-tools/pkg/promotion"
+	"golang.org/x/sync/semaphore"
+	"k8s.io/apimachinery/pkg/util/sets"
+)
+
+const (
+	repoAnnotation = "io.openshift.build.source-location"
+)
+
+// ocpImages is image -> repo
+type ocpImages map[string]string
+
+// getImageFromReleaseImage return version, images->repo map, error
+func getImagesFromReleaseImage(releaseImage string) (string, ocpImages, error) {
+	images := ocpImages{}
+	releaseInfoRaw, err := exec.Command("oc", "adm", "release", "info", releaseImage, "--commits", "--output=json").CombinedOutput()
+	if err != nil {
+		return "", nil, err
+	}
+
+	// TODO: There's probably a struct for this somewhere
```

Comment:
Seems to be here https://github.com/openshift/oc/blob/master/pkg/cli/admin/release/info.go#L680-L703

Reply:
not sure if it's kosher to import the `oc` CLI instead of the API repos, should check if it a) messes with go mod and b) increases out compilation time by an hour before we do it ;)